NeoML/src/Dnn/Layers/DropoutLayer.cpp

```diff
@@ -72,16 +75,15 @@ void CDropoutLayer::SetBatchwise( bool value )
 	if( value != isBatchwise ) {
 		isBatchwise = value;
 		if( GetDnn() != nullptr ) {
-			disableDropoutDesc();
+			destroyDropoutDesc();
+			initDropoutDesc();
 		}
 	}
 }
 
 void CDropoutLayer::OnReshaped()
 {
-	if( desc == nullptr ) {
-		desc = static_cast<CBaseDropoutDesc*>(MathEngine().InitDropout(dropoutRate, isSpatial, isBatchwise));
-	}
```

Comment:
Better move `destroyDropoutDesc();` before `initDropoutDesc();` into `OnReshaped()` and just call `OnReshaped` in setters